tests/jballerina-unit-test/src/test/resources/test-src/query/query-expr-with-query-construct-type.bal

```diff
@@ -522,3 +522,84 @@ function testQueryConstructingTableUpdateKeyPanic2() returns error? {
 
     r2.id = 2;
 }
+
+type CustomErrorDetail record {|
+    string message;
+    int code;
+|};
+
+type CustomError error<CustomErrorDetail>;
+
+function testTableOnConflict() {
+    boolean testPassed = true;
+    error? onConflictError1 = error("Key Conflict", message = "cannot insert.");
+    error|null onConflictError2 = ();
+    error|null onConflictError3 = null;
+    CustomError? onConflictError4 = error ("error msg 1", message = "error 1", code = 500);
+    CustomError? onConflictError5 = error ("error msg 2", message = "error 2", code = 500);
+
+    Customer c1 = {id: 1, name: "Melina", noOfItems: 12};
+    Customer c2 = {id: 1, name: "James", noOfItems: 5};
+    Customer c3 = {id: 3, name: "Anne", noOfItems: 20};
+
+    Customer[] customerList = [c1, c2, c3];
+
+    var customerTable1 = table key(id) from var customer in customerList
+        select {
+            id: customer.id,
+            name: customer.name,
+            noOfItems: customer.noOfItems
+        }
+        on conflict onConflictError1;
+
+    assertEqual(customerTable1, onConflictError1);
+
+    var customerTable2 = table key(id) from var customer in customerList
+        select {
+            id: customer.id,
+            name: customer.name,
+            noOfItems: customer.noOfItems
+        }
+        on conflict onConflictError2;
+
+    assertEqual(customerTable2, table key(id) [{"id":1,"name":"James","noOfItems":5},{"id":3,"name":"Anne","noOfItems":20}]);
+
+    var customerTable3 = table key(id) from var customer in customerList
+        select {
+            id: customer.id,
+            name: customer.name,
+            noOfItems: customer.noOfItems
+        }
+        on conflict onConflictError3;
+
+    assertEqual(customerTable3, table key(id) [{"id":1,"name":"James","noOfItems":5},{"id":3,"name":"Anne","noOfItems":20}]);
+
+    var customerTable4 = table key(id) from var customer in customerList
+        select {
+            id: customer.id,
+            name: customer.name,
+            noOfItems: customer.noOfItems
+        }
+        on conflict onConflictError4;
+
+    assertEqual(customerTable4, onConflictError4);
+
+    var customerTable5 = table key(id) from var customer in customerList
+        select {
+            id: customer.id,
+            name: customer.name,
+            noOfItems: customer.noOfItems
+        }
+        on conflict onConflictError5;
```

Comment:
Shall we add a test case with nil literal as on conflict epression?

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/36341/commits/efd65f8590074d58c24bb72ab56de4b8014c1288